templates/strict_authoritative/bind/named.conf.local.j2

```diff
@@ -0,0 +1,127 @@
+// BIND 9 Configuration - named.conf.local - generated by systemli.bind9 ansible role
+// DO NOT edit, change your ansible config and re-run your ansible playbooks
+//
+
+// Consider adding the 1918 zones here, if they are not used in your
+// organization
+//include "/etc/bind/zones.rfc1918";
+
+{% if bind9_statistics_enabled %}
+
+statistics-channels {
+  inet 127.0.0.1 port 8053 allow { 127.0.0.1; };
+};
+{% endif %}
+{% if bind9_masters is defined or bind9_masters_extra is defined %}
+
+// masters clause lists for masters secondary zones and also-notify statements
+// and ACLs with same name and content, to be used in allow-transfer statements
+{%   for master in bind9_masters | default ([]) + bind9_masters_extra | default ([]) %}
+masters {{ master.name }} {
+{%     for addr in master.addresses %}
+    {{ addr }};
+{%     endfor %}
+};
+acl {{ master.name }} {
+{%     for addr in master.addresses %}
+    {{ addr }};
+{%     endfor %}
+};
+
+{%   endfor %}
+{% endif %}
+{% if bind9_acl is defined %}
+
+	// Custom acls
+{%   for acl_item in bind9_acl %}
+acl {{ acl_item.name }} {
+{%     for acl_address in acl_item.addresses %}
+	{{ acl_address }};
+{%     endfor %}
+};
+{%   endfor %}
+{% endif %}
+
+// The following zones are managed by this DNS Server //
+{% for zone in ( bind9_zones_static + bind9_zones_dynamic ) | sort( attribute='name' ) %}
+{%   set zone_type = zone.type | default( bind9_zone_type | default( 'master' ) ) %}
+zone "{{ zone.name }}" {
+    type {{ zone_type }};
+{# ############################### Type zone specific statements ####################################### #}
+{%   if zone_type == 'master' %}
+    file "/etc/bind/zones/db.{{ zone.name }}";
+{%   elif zone_type == 'slave' %}
+    file "/var/lib/bind/db.{{ zone.name }}";
+{%     if zone.masters is defined or bind9_masters is defined %}
+    masters {
+{%       if zone.masters is defined %}
+{%         for master in zone.masters %}
+        {{ master }};
+{%         endfor %}
+{%       else  %} {# so bind9_masters is defined #}
+{%         for master in bind9_masters %}
+        {{ master.name }};
+{%         endfor %}
+{%       endif %}
+    };
+{%     endif %}
+{%   elif zone_type == 'forward' %}
+    forwarders {
+{%     for fwd in zone.forwarders %}
+        {{ fwd }}; 
+{%     endfor %}
+   }; 
+{%   endif %}
+{# ############################### Common statements to all zone types ####################################### #}
+    allow-query {
+{%   if zone.allow_query is defined and zone.allow_query | length() > 0  %}
+{%     for allow_query_item in zone.allow_query | list %}
+        {{ allow_query_item }};
+{%     endfor %}
+    };
+{%   else %}
+        any;
+    };
+{%   endif %}
+{%   if zone.notify is defined %}
+     notify {{ zone.notify }};
```

Comment:
A filter `| ternary ('yes','no')` is missing.

Reply:
Default templates use this filter and manage separately (and strangely) `explicit` value with code, but not `master-only` that may be interesting to avoid useless notification noise. For new templates I preferred simpler code, with exact values for the clause.